test/blackbox/run_blackbox_tests/test_cases_set_metaconn_retry_cb.c

```diff
@@ -0,0 +1,130 @@
+/*
+    test_cases_set_metaconn_retry_cb.c -- Execution of specific meshlink black box test cases
+    Copyright (C) 2019  Guus Sliepen <guus@meshlink.io>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "execute_tests.h"
+#include "test_cases_set_metaconn_retry_cb.h"
+#include "../common/test_step.h"
+#include "../common/common_handlers.h"
+#include "../../utils.h"
+#include <assert.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <pthread.h>
+
+static void test_case_set_mataconn_retry_cb_01(void **state);
+static bool test_set_mataconn_retry_cb_01(void);
+
+static bool status;
+static int retry_attempts;
+static struct sync_flag join_cond = {.mutex  = PTHREAD_MUTEX_INITIALIZER, .cond = PTHREAD_COND_INITIALIZER};
+
+static void status_cb(meshlink_handle_t *mesh, meshlink_node_t *source, bool reach) {
+	if(!strcmp(mesh->name, "foo") && !strcmp(source->name, "bar")) {
+		status = reach;
+		set_sync_flag(&join_cond, true);
```

Comment:
You're setting the flag even if the node becomes unreachable. Maybe rename `join_cond` to `status_changed_cond`? Also, instead of `status`, name it `bar_reachable`.